tests/integration/ha_tests/helpers.py

```diff
@@ -551,6 +561,7 @@ async def reused_storage(ops_test: OpsTest, unit_name) -> bool:
     await run_command_on_unit(
         ops_test,
         unit_name,
-        "grep 'Database cluster state: in archive recovery' /var/snap/charmed-postgresql/common/var/log/patroni/patroni.log",
+        "grep -E 'Database cluster state: in archive recovery|Database cluster state: shut down' "
```

Comment:
The first cluster's primary seems to be in `Database cluster state: shut down`

Reply:
I tried checking based on role, but it was unstable in CI, so I split the functions based on usage.